### ISSUE TITLE
Add country China

### ISSUE THREAD

**Issue by tinohager** [tinohager/Nager.Date#52]
Source:
https://en.wikipedia.org/wiki/Public_holidays_in_China

Challenge:

5th solar term calculation

- https://stackoverflow.com/questions/8708048/position-of-the-sun-given-time-of-day-latitude-and-longitude
- https://stackoverflow.com/questions/2056555/c-sharp-sunrise-sunset-with-latitude-longitude
- http://www.stjarnhimlen.se/comp/tutorial.html
- https://en.wikipedia.org/wiki/Position_of_the_Sun
- http://guideving.blogspot.co.at/2010/08/sun-position-in-c.html

**Comment by rubenmonteiro** [comment]
The nuget on my pull request my give a hand on calculating this particular holiday

**Comment by tinohager** [comment]
The package already has nearly 30.000 downloads, we have to make sure that all dependencies are license technically correct. Even new packages with 0 downloads are not very trustworthy.

I think it's very nice when we move on. I would prefer if we make a separate pull request for each country we add, so I could better accept individual changes. A pull request with so many changes makes it very difficult.

**Comment by rubenmonteiro** [comment]
[https://github.com/rubenmonteiro/AASharp](https://github.com/rubenmonteiro/AASharp) The fork from the original project and I had to create a nuget because there wasn't any, so it is natural that there are 0 downloads, I made the nuget today. Why try to reinvent the wheel if it is already invented. And the project does all the calculations with no need of searching for formulas and so forth.
Just trying to help

**Comment by tinohager** [comment]
Now two packages available on nuget https://www.nuget.org/packages?q=AASharp

I'm happy about help! I hope we can add many more countries...

**Comment by tinohager** [comment]
I think this project is not only released in binary form so we must check the licence exactly

http://www.naughter.com/aa.html

**Copyright**

- You are allowed to include the source code in any product (commercial, shareware, freeware or otherwise) when your product is released in binary form.
- You are allowed to modify the source code in any way you want except you cannot modify the copyright details at the top of each module.
- If you want to distribute source code with your application, then you are only allowed to distribute versions released by the author. This is to maintain a single distribution point for the source code.

**Comment by rubenmonteiro** [comment]
Sure, but when I forked the project, this sunday, there wasn't any nuget package, now the original github repository that I forked also as a nuget, the readme file indicates an update 4 hours ago that include the link to that nuget [https://www.nuget.org/packages/AASharp](https://www.nuget.org/packages/AASharp) So I suppose that it is better to use that nuget instead of mine because it will be updated more often then mine.

**Comment by tinohager** [comment]
yes I think so too, can you contact the owner of the original project (www.naughter.com/contact.html) if he could possibly change the license to MIT (https://en.wikipedia.org/wiki/MIT_License)

**Comment by rubenmonteiro** [comment]
Will do

**Comment by rubenmonteiro** [comment]
I have contact the owner of the original project and he told me to with the owner of the project that I forked [https://github.com/jsauve/AASharp](https://github.com/jsauve/AASharp)

**Comment by tinohager** [comment]
Great, he's already added a license information, I'll ask if he could change the license to MIT

**Comment by HayriPV** [comment]
Does this help you guys? https://github.com/commenthol/date-chinese
